main.tf

```diff
@@ -383,7 +388,8 @@ module "efs_sg" {
   vpc_id      = local.vpc_id
   description = "Security group allowing access to the EFS storage"
 
-  ingress_cidr_blocks = [var.cidr]
+  ingress_cidr_blocks = local.efs_sg_ingress_cidr
```

Comment:
Looking at this whole security group makes me think that you don't need to allow ingress to the EFS volume from the entire VPC. Instead we can just rely on https://github.com/terraform-aws-modules/terraform-aws-atlantis/blob/master/main.tf#L387-L390, which already allows nfs-tcp from the Atlantis security group.